src/components/GameList/GameList.tsx

```diff
@@ -80,6 +81,13 @@ export class GameList extends React.PureComponent<GameListProps, GameListState>
         }
     };
 
+    isPaused(pause_control: AdHocPauseControl) {
```

Comment:
I might be missing something, and given the compiler isn't complaining there's a good chance I am - but above we have `pause_control?: AdHocPauseControl` indicating it can be `undefined`, which I think is correct. Down below we just pass that field right along to this function, so it seems to me like this needs to also be `pause_control?: AdHocPauseControl` along with a corresponding `undefined` check before we do that `key in pause_control`?

Reply:
> given the compiler isn't complaining

This could just be because strict null checks are off in this repo ([config](https://github.com/online-go/online-go.com/blob/b2448c1fa98213ee2f3e5823e4f80205dd2a1cba/tsconfig.json#L34))

Reply:
Ahh, geeze yeah about time to fix that I reckon, I think this is the last repo that doesn't have that enabled. (Obviously not in this patch, just mental note)

Reply:
Nice catch. Should be fixed in 6567958b04238d4346323d9ff187da96a55468c8.

Reply:
Specifically, here's the error I got (on the other branch):
```
ERROR in ./src/components/GameList/GameList.tsx:99:58
TS1016: A required parameter cannot follow an optional parameter.
     97 |     }
     98 |
  >  99 |     computeRemainingTime(time_control?: JGOFTimeControl, clock: AdHocPlayerClock | number) {
        |                                                          ^^^^^
    100 |         if (!time_control) {
    101 |             return Number.MAX_SAFE_INTEGER;
    102 |         }
```

Reply:
I think the previous logic was correct. I suspect any parameter can be `null` any time, and optional chaining with `?.` just accounts for it. I've force pushed back to the previous state (dropping 6567958), but happy to be overridden (and add it back) if you're confident.

Reply:
(Regardless, I've tested this in cases where `pause_control` is null, so the code works... if there's a type error that'll be exposed when we turn on strict null checks, we can fix it then...)

Reply:
Yep so marks the parameter as optional, and when not provided when called, the default value is `undefined`. All optional parameters need to be at the end of the function, no required parameters after optional so that's what it's complaining about. So the two ways to fix this are to either specify it as `time_control: JGOFTimeControl | undefined` (which differs from `?` in which would essentially expand to `time_control: JGOFTimeControl | undefined = undefined`), or move `clock` to be before the `time_control` parameter.

Reply:
That helps, thanks. I hadn't internalized that `undefined` is an independent type from `null` (although both "null-ish"). I've also now read the typescript docs on `?:` and `?.` (previously I inferring from other languages, context, and compiler errors). I'll send a new version of the fix-up.

> it will ever actually be undefined though so might be fine

`game.json?.pause_control` can be undefined; based on dumps, it's missing from `game.json` for games that have never been paused (after unpausing it's defined but equal to `{}`). A bit confused I didn't see this causing a problem... if `for (key in undefined)` raises a TypeError exception, then never-paused games should have been sorted last (that's where exceptions go), but that's not what I saw... based on observation, I assumed the `for (key in pause_control)` was throwing `undefined` to the `default:` case.

Anyway, I'll get this cleaned up and test again with some fresh data.

Reply:
You know what, `for (key in undefined)` doesn't throw an error, I was testing the wrong thing (`key in undefined`, the key existence check as opposed to an iterator, does throw an error). So, actually it's totally fine as is, though still good to mark the parameter as being able to accept undefined.

Reply:
Aha, that explains it, and good to know. I pushed faafa791e1e17ce2f921d25abb299e531a86aee0 to resolve this.